project/dexter-peer-review/src/java/com/samsung/sec/dexter/peerreview/plugin/DexterPeerReviewPlugin.java

```diff
@@ -0,0 +1,263 @@
+		defect.setClassName("");
+		defect.setMethodName("");
+		defect.setLanguage(config.getLanguageEnum().toString());
+		defect.setSeverityCode(severity);
+		defect.setCategoryName("PeerReview");
+		defect.setMessage("");
+		defect.setToolName(PLUGIN_NAME);
+		defect.setAnalysisType(config.getAnalysisType().name());
+
+		for (DPRComment defectInfo : defectInfos) {
+			Occurence occ = new Occurence();
+			occ.setStartLine(defectInfo.getStartLine());
+			occ.setEndLine(defectInfo.getEndLine());
+			occ.setStringValue("DPR");
+			occ.setMessage(defectInfo.getReviewComment());
+
+			defect.addOccurence(occ);
+		}
+
+		return defect;
+	}
+
+	static String getFullCommentWithoutLineSeparator(String comment) {
+		return comment.replace("\r\n", "").replace("\n", "").trim();
+	}
+
+	private HashMap<String, String> getSeverityAndCommentFromFullComment(String comment) {
+		String severity = "";
+		String reviewComment = "";
+
+		Pattern criticalPattern = Pattern.compile(criticalRegExp);
+		Matcher criticalMatcher = criticalPattern.matcher(comment);
+
+		Pattern majorPattern = Pattern.compile(majorRegExp);
+		Matcher majorMatcher = majorPattern.matcher(comment);
+
+		Pattern defaultPattern = Pattern.compile(simpleRegExp);
+		Matcher defaultMatcher = defaultPattern.matcher(comment);
+
+		if (criticalMatcher.find()) {
+			severity = "CRI";
+			reviewComment = comment.substring(criticalMatcher.end());
+		} else if (majorMatcher.find()) {
+			severity = "MAJ";
+			reviewComment = comment.substring(majorMatcher.end());
+		} else if (defaultMatcher.find()) {
+			severity = "CRC";
+			reviewComment = comment.substring(defaultMatcher.end());
+		}
+
+		HashMap<String, String> infoHashMap = new HashMap<String, String>();
+		infoHashMap.put("severity", severity);
+		infoHashMap.put("comment", reviewComment.trim());
+
+		return infoHashMap;
+	}
+
+	static int getLineFromSourcecode(CharSequence sourcecode, int start) {
+		int line = 1;
+		Pattern pattern = Pattern.compile("\n");
+		Matcher matcher = pattern.matcher(sourcecode);
+		matcher.region(0, start);
+		while (matcher.find()) {
+			line++;
+		}
+		return (line);
+	}
+
+	@Override
+	public boolean supportLanguage(LANGUAGE language) {
+		if (language == LANGUAGE.C || language == LANGUAGE.CPP || language == LANGUAGE.JAVA) {
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public String[] getSupportingFileExtensions() {
+		return new String[] { "c", "cpp", "h", "hpp", "java" };
+	}
+
+	public ArrayList<DPRComment> getAllDPRCommentFromSourcecode(CharSequence sourcecode) {
+		final String regExp = "(?:/\\*.*[w]*(DPR:)(?:[^*]|(?:\\*+[^*/]))*\\*+/)|(?://.*[w]*(DPR:).*)";
+		ArrayList<DPRComment> matchList = new ArrayList<>();
+
+		Pattern pattern = Pattern.compile(regExp);
+		Matcher matcher = pattern.matcher(sourcecode);
+
+		while (matcher.find()) {
+			DPRComment comment = new DPRComment();
+			comment.setStartLine(getLineFromSourcecode(sourcecode, matcher.start()));
```

Comment:
This code seems be different from what we talked about during the code-reivew meeting.
Did we decide to calculate the line only once?